File: packages-gen/dart/ui/lib/state/dev.dart
create(enabled, capabilities) {
  return <dynamic, dynamic>{
    "enabled":true == enabled,
    "capabilities":capabilities ?? <dynamic, dynamic>{},
    "status":"idle",
    "summary":<dynamic, dynamic>{},
    "panels":<dynamic, dynamic>{}
  };
}

availablep(state, capability_id) {
  return (true == state["enabled"]) && (true == state["capabilities"][capability_id]);
}

set_summaryf(state, summary) {
  state["summary"] = (summary ?? <dynamic, dynamic>{});
  state["status"] = "ready";
  return state;
}

File: packages-gen/js/libs/xt-ui/state/dev.js
function create(enabled,capabilities){
  return {
    "enabled":true == enabled,
    "capabilities":capabilities || {},
    "status":"idle",
    "summary":{},
    "panels":{}
  };
}

function availablep(state,capability_id){
  return (true == state["enabled"]) && (true == state["capabilities"][capability_id]);
}

function set_summaryf(state,summary){
  state["summary"] = (summary || {});
  state["status"] = "ready";
  return state;
}

module.exports = {
  ["create"]:create,
  ["availablep"]:availablep,
  ["set_summaryf"]:set_summaryf
}